server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -332,4 +332,22 @@ public int compareTo(ByteSizeValue other) {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(toString());
     }
+
+    /**
+     * @return Constructs a {@link ByteSizeValue} with the bytes resulting from the addition of the arguments' bytes. Note that the
+     *         resulting {@link ByteSizeUnit} is bytes.
+     * @throws ArithmeticException if the resulting bytes overflow a long
+     */
+    public static ByteSizeValue add(ByteSizeValue x, ByteSizeValue y) {
+        return ByteSizeValue.ofBytes(Math.addExact(x.getBytes(), y.getBytes()));
+    }
+
+    /**
+     * @return Constructs a {@link ByteSizeValue} with the bytes resulting from the difference of the arguments' bytes. Note that the
+     *         resulting {@link ByteSizeUnit} is bytes.
+     * @throws ArithmeticException if the resulting bytes overflow a long
```

Comment:
I do not think this can ever happen (once `-1` is handled). But we would want to ensure the result is not `< 0`. In particular, we should check `-1`, since that will pass `ByteSizeValue.ofBytes`.

Reply:
Sure thing. We can throw an ArithmeticException for the case that the result is -1. Please see the relevant commit in my next push and unresolve this conversation if it's not sufficient. Thanks!

Reply:
I think we should just throw `IllegalArgumentException`. It is strange to report different errors for -1 and other negative values. And `ByteSizeValue.ofBytes(-2)` will throw `IllegalArgumentException`.

Reply:
@henningandersen , done. Please check newly pushed commits.